.github/workflows/rhbop_productized_pull_request.yml

```diff
@@ -0,0 +1,54 @@
+# Tests RHBOP productized profile builds on PRs
+name: Productized Build Chain
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, labeled]
+    branches:
+      - main
+      - 8.*
+    paths-ignore:
+      - 'LICENSE*'
+      - '.gitignore'
+      - '**.md'
+      - '**.adoc'
+      - '*.txt'
+      - '.ci/jenkins/**'
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  productized-build-chain:
+    concurrency:
+      group: rhbop_pull_request-${{ github.event_name }}-${{ github.head_ref }}-${{ matrix.os }}-${{ matrix.java-version }}
+      cancel-in-progress: true
+    timeout-minutes: 120
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        java-version: [11]
+        maven-version: ['3.8.6']
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    name: ${{ matrix.os }} - Java ${{ matrix.java-version }} - Maven
+    steps:
+      - name: Clean Disk Space
+        uses: kiegroup/kogito-pipelines/.ci/actions/ubuntu-disk-space@main
+      - name: Java and Maven Setup
+        uses: kiegroup/kogito-pipelines/.ci/actions/maven@main
+        with:
+          java-version: ${{ matrix.java-version }}
+          maven-version: ${{ matrix.maven-version }}
+          cache-key-prefix: ${{ runner.os }}-${{ matrix.java-version }}-maven${{ matrix.maven-version }}
```

Comment:
Why not `${{ matrix.os }}`; is there a difference?

Reply:
In this specific context I don't think there would be much difference on using either `matrix.os` or `runner.os`. `matrix.os` is the actual os, e.g., `ubuntu` whereas `runner.os` is the os of the runner executing the job, whose possible values are Linux, Windows, or macOS, in this case it would be `linux`. Personally I would keep `runner.os` as we are doing in [pull_request.yml](https://github.com/kiegroup/optaplanner/blob/main/.github/workflows/pull_request.yml#L68).

Reply:
I would keep `runner.os` too since `Java and Maven Setup` step is more reusable no matter the os is fixed on or is part of a matrix